gdnative-core/src/geom/basis.rs

```diff
@@ -39,6 +106,201 @@ impl Basis {
         }
     }
 
+    pub fn determinant(&self) -> f32 {
+        let elements = self.elements;
+        return elements[0].x * (elements[1].y * elements[2].z - elements[2].y * elements[1].z)
+            - elements[1].x * (elements[0].y * elements[2].z - elements[2].y * elements[0].z)
+            + elements[2].x * (elements[0].y * elements[1].z - elements[1].y * elements[0].z);
+    }
+
+    fn orthonormalize(&mut self) {
+        debug_assert_ne!(self.determinant(), 0.0, "Determinant should not be zero.");
+
+        // Gram-Schmidt Process
+        let mut x = self.x();
+        let mut y = self.y();
+        let mut z = self.z();
+
+        x = x.normalize();
+        y = y - x * (x.dot(y));
+        y = y.normalize();
+        z = z - x * (x.dot(z)) - y * (y.dot(z));
+        z = z.normalize();
+
+        self.set_x(x);
+        self.set_y(y);
+        self.set_z(z);
+    }
+
+    pub fn orthonormalized(&self) -> Basis {
+        let mut c = *self;
+        c.orthonormalize();
+        c
+    }
+
+    pub fn is_equal_approx(&self, other: &Basis) -> bool {
+        return self.elements[0].approx_eq(&other.elements[0])
+            && self.elements[1].approx_eq(&other.elements[1])
+            && self.elements[2].approx_eq(&other.elements[2]);
+    }
+
+    fn is_orthogonal(&self) -> bool {
+        let identity = Self::identity();
+        let m = (*self) * self.transposed();
+        m.is_equal_approx(&identity)
+    }
+
+    fn is_rotation(&self) -> bool {
+        let det = self.determinant();
+        det.approx_eq(&1.0) && self.is_orthogonal()
+    }
+
+    fn get_quat(&self) -> Quat {
+        debug_assert!(!self.is_rotation(), "Basis must be normalized in order to be casted to a Quaternion. Use get_rotation_quat() or call orthonormalized() instead.");
+
+        /* Allow getting a quaternion from an unnormalized transform */
```

Comment:
A normal comment should be used instead.

Reply:
Updated